### PR TITLE
Validating difi

### LINKED ISSUE #324: Demo notebook for Difi
Essentially will step through individual parts of function and make sure they work as expected: if there are only single/double observations of an object, no linking should be performed, for example.

### LINKED ISSUE #449: Change "window_summary" to _ in PPLinkingFilter
We aren't using the window_summary variable, so there's no point creating it - line 85 of PPLinkingFilter.

### PR DESCRIPTION
**Pull request by astronomerritt**
Fixes #324.
Fixes #449.

There is now a validation demo notebook for PPLinkingFilter in `docs/notebooks/`. It demonstrates both successful linking of a test object, and unsuccessful linking based on each of four different criteria being unfulfilled: not enough tracklets/observations for a tracklet, not enough tracklets in the time window, tracklet observations too close together spatially, and subsequent tracklet observations too far away in time. It also demonstrates detection efficiency.

I updated README.md in `docs/notebooks/` as well to include the new notebook.

I have also rewritten test_PPLinkingFilter to be more useful based on the notebook. Unfortunately as a side effect it takes quite a lot longer but it does check everything that the notebook checks.

I also replaced the unused variable `window_summary` in PPLinkingFilter with an _.

## Review Checklist for Source Code Changes

- [x] Does pip install still work?
- [x] Have you written a unit test for any new functions?
- [x] Do all the units tests run successfully?
- [x] Does Sorcha run successfully on a test set of input files/databases?
- [x] Have you used black on the files you have updated to confirm python programming style guide enforcement?

### CODE REVIEW

File: docs/notebooks/README.md
Comment (drewoldag) [inline docs/notebooks/README.md:1]: There's one more place to make an update to get the notebook to render correctly in the documentation: 
https://github.com/dirac-institute/sorcha/blob/main/docs/notebooks.rst 

Add a line in the table of contents like:
`Linking Filter Demo <notebooks/demo_LinkingFilter>`

You'll probably also need to include a markdown cell at the top of the notebook that will be used as a title. 
See this notebook as an example: https://github.com/dirac-institute/sorcha/blob/main/docs/notebooks/demo_ApparentMagnitudeValidation.ipynb

**Review by drewoldag — APPROVED** [review]
Looks good to me. I left a couple of instruction comments for how to get the notebook rendered on RTD.

File: docs/notebooks/README.md
Comment (astronomerritt) [reply docs/notebooks/README.md:1]: Done both of these - also merged main back into my branch, so the CI is failing for the same reasons it was failing earlier. I will leave merging of this PR up to you if you think it is appropriate :)

File: tests/sorcha/test_PPLinkingFilter.py [hunk 1]
```diff
@@ -1,34 +1,131 @@
 import pandas as pd
 import numpy as np
-from numpy.testing import assert_equal
 
 from sorcha.utilities.dataUtilitiesForTests import get_test_filepath
 
 
 def test_PPLinkingFilter():
```
Comment (moeyensj) [inline tests/sorcha/test_PPLinkingFilter.py:7]: These tests look really good to me! Good sanity checks of some of the parameters in difi.

**Review by moeyensj — APPROVED** [review]
Looks good to me! I'd be interested in learning more about the performance and if I can help provide more optimizations.

**Pull request merged**